sentry_sdk/tracing.py

```diff
@@ -130,7 +130,16 @@ def __init__(
         start_timestamp=None,  # type: Optional[datetime]
     ):
         # type: (...) -> None
-        self.trace_id = trace_id or uuid.uuid4().hex
+        if trace_id:
+            self.trace_id = trace_id
+        elif hub:
+            traceparent = hub.get_traceparent()
```

Comment:
the constructor already takes an explicit `trace_id`, so the hub based logic should be in `hub.start_span` and not here. Please leave this constructor unchanged.